plugins/catalog-backend-module-bitbucket-server/src/lib/BitbucketServerClient.ts

```diff
@@ -28,6 +29,7 @@ import { BitbucketServerProject, BitbucketServerRepository } from './types';
  */
 export class BitbucketServerClient {
```

Comment:
It's kinda nice to implement this functionality within this client directly, but right now there's a bit of a trap with that in that most implementations create this client themselves. That means that the actual number of requests hitting the Bitbucket deployment might be significantly higher than the configured throttling. In the end I think it might be best to implement this in the Bitbucket discovery provider instead? it also reduces the risk that this negatively impacts requests in the user path, like scaffolder actions.

Reply:
With this suggestion, would the throttling config options be moved to the discovery provider as well?

Reply:
Yep I think that would be the best place to put them to avoid confusion

Reply:
Ok, I've switched over to p-throttle. I have also moved the throttling logic into the entity provider as well. I believe this should be ready for another look.